Improved text of exception message in case of error in module's composer.json

vovayatsyuk opened the pull request:
### Description

When some third-party module has syntax error inside its composer.json, it's very hard to determine which
module caused an exception. 

This PR changes default message with the following:

```
Some_Module's composer.json error: Decoding failed: Syntax error
```

where `Decoding failed: Syntax error` - is a message of the original exception

### Manual testing scenarios
<!--- Provide a set of unambiguous steps to test the proposed code change -->
> Sample data must be installed to test this behavior.
1. Make an error in `app/code/Magento/Cms/composer.json`:
2. Run `php bin/magento sampeladata:reset`. You'll receive a message:

    ```
    [Zend_Json_Exception]          
    Decoding failed: Syntax error
    ```

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

vovayatsyuk committed:
Improved text of exception message

When some third-party module has syntax error inside its composer.json, it's very hard to determine which
module caused an exception. 

This PR changes default message with the following:

```
Some_Module's composer.json error: Decoding failed: Syntax error
```

adragus-inviqa commented:
I wouldn't concat any other string with the module name; it'd keep it completely separate, to make it more obvious.
Something like`Error decoding composer.json for module Some_Module`.

vovayatsyuk commented:
I'd like to keep original exception message without change, because there are multiple possible variants for it. So we need a template that will suite for all kind of messages.

vovayatsyuk committed:
Improved code readability

- Original exception message moved into 'sprintf' arguments
- "'s" suffix removed from module name

adragus-inviqa commented:
I'm not sure what you mean by "a template that will suite for all kind of messages"; it just has to be a good format. 

But your recent change looks good. I just didn't like the interpolation: `$moduleName's`.

vovayatsyuk committed:
Trailing spaces removed

magento-team commented:
@vovayatsyuk thank you for your contribution. Your Pull Request has been successfully merged
